src/sentry/runner/commands/configoptions.py

```diff
@@ -0,0 +1,150 @@
+        click.echo(f"{opt.name}: {options.get(opt.name)}")
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def patch(filename, dryrun):
+    "Updates, gets, and deletes options that are each subsectioned in the given file."
+    import yaml
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+        data = configmap_data.get("data", {}).get("options-patch.yaml", "")
+
+        keysToFetch = data.get("fetch", {})
+        keysToFetch = (line for line in keysToFetch.split("\n") if line)
+
+        keysToUpdate = data.get("update", {})
+        keysToUpdate = (line.split(": ") for line in keysToUpdate.split("\n") if line)
+
+        keysToDelete = data.get("delete", {})
+        keysToDelete = keysToFetch = (line for line in keysToDelete.split("\n") if line)
+
+        for key in keysToFetch:
+            click.echo(get(key, dryrun))
+
+        for key, val in keysToUpdate.items():
+            click.echo(set(key, dryrun))
+
+        for key in keysToDelete:
+            click.echo(delete(key, dryrun))
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def strict(filename, dryrun):
+    "Deletes everything not in the uploaded file, and applies all of the changes in the file."
+    import yaml
+
+    from sentry import options
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+
+        data = configmap_data.get("data", {}).get("options-strict.yaml", "")
+
+        kv_generator = (line.split(": ") for line in data.split("\n") if line)
+
+        db_keys = (opt.name for opt in options.all())
+
+        # update the database to remove all keys NOT in the given file.
+        for key in db_keys:
+            if key not in kv_generator:
+                click.echo(delete(key, dryrun))
+
+        for line in data.split("\n"):
+            if line:
+                line_tuple = line.split(": ")
+                key = line_tuple[0]
+                val = ""
+                if len(line_tuple) > 1:
+                    val = line_tuple[1]
+                click.echo(set(key, val, dryrun))
+
+
+def get(key, dryrun=False):
+    from sentry import options
+    from sentry.options.manager import UnknownOption
+
+    try:
+        opt = options.lookup_key(key)
+        return f"Fetched Key: {opt.name} ({opt.type}) = {options.get(opt.name)}"
+    except UnknownOption:
+        return "unknown option: %s" % key
```

Comment:
For the same reasoning above, please bubble up the exception instead of turning it into a string.
The language provides us a structured way to communicate error conditions between methods through exceptions. You can leverage this mechanism and let the entry point of the application decide what to do with those exceptions. The entry point (or anything on the presentation side) is the right place to decide how to present the error to the user as it is the only place where we have context on how the interaction with the user happens.

Reply:
👍